packages/frontend/src/components/Explorer/FiltersCard/FiltersCard.tsx

```diff
@@ -35,9 +40,66 @@ const FiltersCard: FC = memo(() => {
     const tableName = useExplorerContext(
         (context) => context.state.unsavedChartVersion.tableName,
     );
-    const filters = useExplorerContext(
-        (context) => context.state.unsavedChartVersion.metricQuery.filters,
-    );
+    const { data } = useExplore(tableName);
+
+    const refreshRequiredFiltersProperty = (
+        allFilter: FilterGroup | undefined,
+    ) => {
+        if (!allFilter) return;
+        if (data && data.tables[tableName]) {
+            const requiredFilters =
+                data.tables[tableName].required_filters || [];
+            const allRequiredFilters: FilterRule[] =
+                reduceRequiredDimensionFiltersToFilterRules(
+                    requiredFilters,
+                    data.tables[tableName],
+                    undefined,
+                );
+            const allFilterRefs = allRequiredFilters.map(
+                (filter) => filter.target.fieldId,
+            );
+            resetRequiredFilterRules(allFilter, allFilterRefs);
+        }
+    };
+    const updateDimensionFiltersWithRequiredFilters = (
+        unsavedQueryFilters: Filters,
+    ) => {
+        if (data && data.tables[tableName]) {
+            const requiredFilters = data.tables[tableName].required_filters;
+            if (requiredFilters && requiredFilters.length > 0) {
+                const reducedRules: FilterRule[] =
+                    reduceRequiredDimensionFiltersToFilterRules(
+                        requiredFilters,
+                        data.tables[tableName],
+                        unsavedQueryFilters.dimensions,
+                    );
+                unsavedQueryFilters.dimensions =
+                    overrideFilterGroupWithFilterRules(
+                        unsavedQueryFilters.dimensions,
+                        reducedRules,
+                    );
+            }
+        }
+        return unsavedQueryFilters;
+    };
+    const resetDimensionFiltersIfNoModelSelected = (
+        unsavedQueryFilters: Filters,
+    ) => {
+        if (tableName.length === 0) {
```

Comment:
can you explain this pls? 😅

Reply:
Sure! i had to add this code, as on the ui page when you are creating a chart/query is you selected a different model, or move back and forth between models, the required filters need to be cleaned out, as different models have different filters.